script/foundry/utils/PrecompileTest.sol

```diff
@@ -0,0 +1,451 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import { Script } from "forge-std/Script.sol";
+import { console2 } from "forge-std/console2.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { PILTerms, IPILicenseTemplate } from "../../../contracts/interfaces/modules/licensing/IPILicenseTemplate.sol";
+import { IIPAssetRegistry } from "../../../contracts/interfaces/registries/IIPAssetRegistry.sol";
+import { ILicensingModule } from "../../../contracts/interfaces/modules/licensing/ILicensingModule.sol";
+import { IRoyaltyModule } from "../../../contracts/interfaces/modules/royalty/IRoyaltyModule.sol";
+import { IIpRoyaltyVault } from "../../../contracts/interfaces/modules/royalty/policies/IIpRoyaltyVault.sol";
+import { IIPAccount } from "../../../contracts/interfaces/IIPAccount.sol";
+import { IVaultController } from "../../../contracts/interfaces/modules/royalty/policies/IVaultController.sol";
+import { IGraphAwareRoyaltyPolicy } from "../../../contracts/interfaces/modules/royalty/policies/IGraphAwareRoyaltyPolicy.sol";
+
+import { MockERC721 } from "../../../test/foundry/mocks/token/MockERC721.sol";
+import { MockExternalRoyaltyPolicy1 } from "../../../test/foundry/mocks/policy/MockExternalRoyaltyPolicy1.sol";
+
+contract PrecompileTest is Script {
+    // user
+    address internal USER = 0x304D6EAA54AD10aF8E98F980aC34ca667F8eB9fd;
+
+    // protocol addresses
+    address internal ROYALTY_POLICY_LAP = 0x28b4F70ffE5ba7A26aEF979226f77Eb57fb9Fdb6;
+    address internal ROYALTY_POLICY_LRP = 0x7D2d9af4E4ab14Afcfd86436BC348928B40963Dd;
+    address internal SUSD = 0x12A8b0DcC6e3bB0915638361D9D49942Da07F455;
+    address internal PIL_TEMPLATE = 0x58E2c909D557Cd23EF90D14f8fd21667A5Ae7a93;
+    address internal IP_ASSET_REGISTRY = 0x28E59E91C0467e89fd0f0438D47Ca839cDfEc095;
+    address internal LICENSING_MODULE = 0x5a7D9Fa17DE09350F481A53B470D798c1c1aabae;
+    address internal ROYALTY_MODULE = 0xEa6eD700b11DfF703665CCAF55887ca56134Ae3B;
+
+    // terms
+    uint256 internal mintingFee = 1000e18;
+    uint256 internal commDerivTermsIdLap10;
+    uint256 internal commDerivTermsIdLap10_NoReciprocal;
+    uint256 internal commDerivTermsIdLap1;
+    uint256 internal commDerivTermsIdLrp10;
+    uint256 internal commDerivTermsIdLrp10_NoReciprocal;
+    uint256 internal commDerivTermsIdExt10;
+
+    // others
+    MockERC721 mockNft;
+    MockExternalRoyaltyPolicy1 mockExternalRoyaltyPolicy;
+    mapping(uint256 tokenId => address ipAccount) internal ipAcct;
+
+    function run() public {
+        vm.startBroadcast();
```

Comment:
Normally it starts broadcast with wallet private key.
``` 
privateKey = vm.envUint("STORY_PRIVATEKEY");
vm.startBroadcast(privateKey);
````

Reply:
addressed in https://github.com/storyprotocol/protocol-core-v1/pull/262/commits/c6587f31ed5f40de369a3e412a79c0fce4af6a57